django/django_test.go

```diff
@@ -349,65 +349,251 @@ func Benchmark_Django(b *testing.B) {
 	engine.AddFunc("isAdmin", func(user string) bool {
 		return user == admin
 	})
-
-	var buf bytes.Buffer
-	var err error
+	require.NoError(b, engine.Load())
 
 	b.Run("simple", func(bb *testing.B) {
 		bb.ReportAllocs()
 		bb.ResetTimer()
+		var buf bytes.Buffer
 		for i := 0; i < bb.N; i++ {
 			buf.Reset()
-			err = engine.Render(&buf, "simple", map[string]interface{}{
+			//nolint:gosec,errcheck // Return value not needed for benchmark
+			_ = engine.Render(&buf, "simple", map[string]interface{}{
 				"Title": "Hello, World!",
 			})
 		}
-
-		require.NoError(b, err)
-		require.Equal(b, expectSimple, trim(buf.String()))
 	})
 
 	b.Run("extended", func(bb *testing.B) {
 		bb.ReportAllocs()
 		bb.ResetTimer()
+		var buf bytes.Buffer
 		for i := 0; i < bb.N; i++ {
 			buf.Reset()
-			err = engine.Render(&buf, "extended", map[string]interface{}{
+			//nolint:gosec,errcheck // Return value not needed for benchmark
+			_ = engine.Render(&buf, "extended", map[string]interface{}{
 				"User": admin,
 			}, "layouts/main")
 		}
-
-		require.NoError(b, err)
-		require.Equal(b, expectExtended, trim(buf.String()))
 	})
 
 	b.Run("simple_with_invalid_binding_keys", func(bb *testing.B) {
 		bb.ReportAllocs()
 		bb.ResetTimer()
+		var buf bytes.Buffer
 		for i := 0; i < bb.N; i++ {
 			buf.Reset()
-			err = engine.Render(&buf, "simple", map[string]interface{}{
+			//nolint:gosec,errcheck // Return value not needed for benchmark
+			_ = engine.Render(&buf, "simple", map[string]interface{}{
 				"Title":       "Hello, World!",
 				"Invalid_Key": "Don't return error from checkForValidIdentifiers!",
 			})
 		}
-
-		require.NoError(b, err)
-		require.Equal(b, expectSimple, trim(buf.String()))
 	})
 
 	b.Run("extended_with_invalid_binding_keys", func(bb *testing.B) {
 		bb.ReportAllocs()
 		bb.ResetTimer()
+		var buf bytes.Buffer
 		for i := 0; i < bb.N; i++ {
 			buf.Reset()
-			err = engine.Render(&buf, "extended", map[string]interface{}{
+			//nolint:gosec,errcheck // Return value not needed for benchmark
+			_ = engine.Render(&buf, "extended", map[string]interface{}{
 				"User":        admin,
 				"Invalid_Key": "Don't return error from checkForValidIdentifiers!",
 			}, "layouts/main")
 		}
+	})
+
+	b.Run("simple_asserted", func(bb *testing.B) {
```

Comment:
I wonder if these benchmarks could be put in a common test package to ensure consistency

Reply:
Thought about that but some are different for some of the engines. The expected values are also different